src/Veldrid/Vk/VkGraphicsDevice.cs

```diff
@@ -1,1519 +1,1523 @@
+                    case VkFence fence:
+                        SetDebugMarkerName(VkDebugReportObjectTypeEXT.FenceEXT, fence.DeviceFence.Handle, name);
+                        break;
+                    case VkSwapchain sc:
+                        SetDebugMarkerName(VkDebugReportObjectTypeEXT.SwapchainKHREXT, sc.DeviceSwapchain.Handle, name);
+                        break;
+                    default:
+                        break;
+                }
+            }
+        }
+
+        private void SetDebugMarkerName(VkDebugReportObjectTypeEXT type, ulong target, string name)
+        {
+            Debug.Assert(_setObjectNameDelegate != null);
+
+            VkDebugMarkerObjectNameInfoEXT nameInfo = VkDebugMarkerObjectNameInfoEXT.New();
+            nameInfo.objectType = type;
+            nameInfo.@object = target;
+
+            int byteCount = Encoding.UTF8.GetByteCount(name);
+            byte* utf8Ptr = stackalloc byte[byteCount + 1];
+            fixed (char* namePtr = name)
+            {
+                Encoding.UTF8.GetBytes(namePtr, name.Length, utf8Ptr, byteCount);
+            }
+            utf8Ptr[byteCount] = 0;
+
+            nameInfo.pObjectName = utf8Ptr;
+            VkResult result = _setObjectNameDelegate(_device, &nameInfo);
+            CheckResult(result);
+        }
+
+        private void CreateInstance(bool debug, VulkanDeviceOptions options)
+        {
+            HashSet<string> availableInstanceLayers = new HashSet<string>(EnumerateInstanceLayers());
+            HashSet<string> availableInstanceExtensions = new HashSet<string>(GetInstanceExtensions());
+
+            VkInstanceCreateInfo instanceCI = VkInstanceCreateInfo.New();
+            VkApplicationInfo applicationInfo = new VkApplicationInfo();
+            applicationInfo.apiVersion = new VkVersion(1, 0, 0);
+            applicationInfo.applicationVersion = new VkVersion(1, 0, 0);
+            applicationInfo.engineVersion = new VkVersion(1, 0, 0);
+            applicationInfo.pApplicationName = s_name;
+            applicationInfo.pEngineName = s_name;
+
+            instanceCI.pApplicationInfo = &applicationInfo;
+
+            StackList<IntPtr, Size64Bytes> instanceExtensions = new StackList<IntPtr, Size64Bytes>();
+            StackList<IntPtr, Size64Bytes> instanceLayers = new StackList<IntPtr, Size64Bytes>();
+
+            if (availableInstanceExtensions.Contains(CommonStrings.VK_KHR_SURFACE_EXTENSION_NAME))
+            {
+                _surfaceExtensions.Add(CommonStrings.VK_KHR_SURFACE_EXTENSION_NAME);
+            }
+
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            {
+                if (availableInstanceExtensions.Contains(CommonStrings.VK_KHR_WIN32_SURFACE_EXTENSION_NAME))
+                {
+                    _surfaceExtensions.Add(CommonStrings.VK_KHR_WIN32_SURFACE_EXTENSION_NAME);
+                }
+            }
+            else if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux)

```

Comment:
We may want to change the use of `RuntimeInformation` throughout the whole library at once. The solution here is not future-proof (checking for NET5_0 will not work for NET6).
We may just want to create a NET6 branch of Veldrid 🤔

Reply:
`OperatingSystem.IsAndroid()` was introduced in net5, and works as expected in net6.
In other words: net5 added, net6 implemented.

Reply:
Well I meant the #if directive in particular. I'm not sure the `NET5_0` symbol will be available in NET6 (the `NET` symbol should exist though)

Reply:
`NET5_0_OR_GREATER`